src/test/java/io/github/oliviercailloux/TestMap.java

```diff
@@ -29,8 +32,9 @@ public static void main(String[] args) throws FileNotFoundException {
 
 		MapFile map = new MapFile(fichier);
 		LOGGER.debug("MapFile", map);
-
-		System.out.println("done");
+		end = "done";
 		map.close();
+		assertEquals(end,"done");
```

Comment:
- BP resources : une ressource dans un jar n’est pas convertible en Fichier. Il faut se contenter d’un stream.
- Votre test devrait tester le contenu de la carte. (Tester que `end` égale `"done"` juste après l’avoir affecté à `"done"`, c’est se moquer de votre lecteur.)

Reply:
Ce test a été réalisé par l'ancien groupe. Ce n'était même pas un test à la base. Je l'ai uniquement transformé en test. Si quelqu'un se moque du lecteur, je ne pense pas que cela soit moi.

Reply:
@oliviercailloux je ne comprends votre remarque première remarque.

Reply:
Utilsation de get.Path après application de la bp ressource

Reply:
- Ne pas accepter `File` en paramètre mais un `InputStream`, plus général : certains flux ne correspondent pas à des fichiers
- L’ancien groupe s’est lui aussi moqué du lecteur, je vous l’accorde (et bien plus que vous). C’est certes une circonstance atténuante, mais ce n’est pas une très bonne excuse. Si vous affirmez transformer leur méthode en un test, il faut qu’il teste quelque chose. Si vous ne désirez pas le faire, alors c’est erroné de prétendre le faire. Dans ce deuxième cas, je suggère de simplement indiquer un commentaire en Javadoc indiquant TODO make a real test out of this. En tous cas, vous pouvez effacer le `assertEquals` qui ne sert à rien, pour ne pas laisser croire erronément que ce test teste effectivement quelque chose.